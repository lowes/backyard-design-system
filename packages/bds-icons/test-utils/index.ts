import { customRender } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }
