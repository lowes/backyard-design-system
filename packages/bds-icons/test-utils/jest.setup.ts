import '@testing-library/jest-dom'
import 'jest-styled-components'

jest.setTimeout(1000 * 60 * 20)
