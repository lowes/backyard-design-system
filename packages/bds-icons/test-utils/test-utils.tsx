import * as React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import * as theme from '@lowes-tech/bds-tokens/v2/light/theme'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
const customRender = (ui, options) => render(ui, { wrapper: RenderWrapper, ...options })

export { customRender }
