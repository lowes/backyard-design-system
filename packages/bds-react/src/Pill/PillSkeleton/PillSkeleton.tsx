import * as React from 'react'
import classnames from 'classnames'
import type { BDSForwardRef } from '../../utils/typings/BDSComponentProps'

import Skeleton from '../../Skeleton'

/**
 * Backyard's PillSkeleton component
 * 
 * Returns a Skeleton circle with predefined heights and widths related
 * to the value of the variant prop (ie. 'pill', 'indicator').
 * 
 * <PillSkeleton />
 */

const PillSkeleton: BDSForwardRef<PillSkeletonProps> = React.forwardRef<
    PillSkeletonRef,
    PillSkeletonProps
>(function PillSkeleton (
    {
        animate = false,
        variant = 'pill',
        size = 'medium',
        className,
        ...props
    },
    ref,
) {

    let height
    let width

    if(size === 'medium') {
        width = (variant === 'pill') 
            ? 'calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2))'
            : 'var(--bds-sizes-size-18)'
        height = (variant === 'pill')
            ? 'calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2))'
            : 'var(--bds-sizes-size-18)'
    } else {
        width = (variant === 'pill')
            ? 'calc(var(--bds-sizes-size-40) + var(--bds-sizes-size-2))'
            : 'calc(var(--bds-sizes-size-32) + var(--bds-sizes-size-2))'
        height = (variant === 'pill')
            ? 'calc(var(--bds-sizes-size-40) + var(--bds-sizes-size-2))'
            : 'calc(var(--bds-sizes-size-32) + var(--bds-sizes-size-2))'
    }

    return (
        <Skeleton 
            variant='circle' 
            width={height} 
            height={width}
            animate={animate}
            className={classnames(className, 'pill--skeleton', `variant--${variant}`)}
            {...props} 
            ref={ref} 
        />
    )
})

type PillSkeletonRef = HTMLDivElement

interface PillSkeletonProps {
    animate?: boolean
    variant?: string
    className?: string
    size?: 'medium' | 'jumbo'
}

PillSkeleton.bdsName = 'PillSkeleton'

export { PillSkeleton }

export type { PillSkeletonProps, PillSkeletonRef }

export default PillSkeleton