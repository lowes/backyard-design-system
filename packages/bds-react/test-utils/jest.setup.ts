import * as React from 'react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

jest.setTimeout(1000 * 60 * 20)

// Mock `useRandomId` hook
// gives consistent ids instead
// of randomly generated ones for
// snapshot tests
let mockId = 0

jest.mock('../src/utils/hooks/useRandomId', () => {
    return jest.fn((id) => id || `${mockId++}`)
})

// Before each test...
beforeEach(() => {
    // Reset `useRandomId` counter
    mockId = 0
})
