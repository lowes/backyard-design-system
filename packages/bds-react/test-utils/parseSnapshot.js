const parseSnapshot = (content) => {
    const data = get_data(content)

    return Object.keys(data).reduce(
        (parsed, key) => ({ ...parsed, [key]: parse(data[key]) }),
        {},
    )
}

function parse(value) {
    const reverted_value = revert_extra_line_breaks(value).replace(/\n/g, '\\n')

    try {
        return reverted_value === 'undefined'
            ? undefined
            : JSON.parse(reverted_value)
    } catch (error) {
        // istanbul ignore next
        throw new Error(
            `Unexpected error while parsing:\n\n>>> start >>>\n${reverted_value}\n<<< end <<<\n\n${
            error.message
            }`,
        )
    }
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L128
function revert_extra_line_breaks(str) {
    return str.includes('\n') ? str.slice(1, -1) : str
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L98
function get_data(content) {
    const data = Object.create(null)
    const populate = new Function('exports', content)
    
    populate(data)
    
    return data
}

export {
    parseSnapshot
}

export default parseSnapshot
