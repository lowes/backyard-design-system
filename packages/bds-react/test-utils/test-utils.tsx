import * as React from 'react'
import { render } from '@testing-library/react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/ThemeProvider'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme="light">{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
// @ts-ignore
const customRender = (ui, options = {}) => render(ui, { wrapper: RenderWrapper, ...options })

/**
 * Function to override custom mount with
 *
 * @param {*} ui
 * @param {*} options
 */
const customMount = (ui, options = {}) =>
    mount(ui, {
        wrappingComponent: ThemeProvider,
        wrappingComponentProps: { theme: 'light' },
        ...options,
    })

export { customRender, customMount }
