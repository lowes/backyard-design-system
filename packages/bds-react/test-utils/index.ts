// @ts-ignore
import { customRender, customMount } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }

// Override shallow method
export { customMount as mount }
