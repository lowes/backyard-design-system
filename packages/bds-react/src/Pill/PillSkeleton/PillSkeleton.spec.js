import * as React from 'react'
import { render } from '../../../test-utils'
import PillSkeleton from './PillSkeleton'

describe('Pill Snapshots', () => {
    test('Pill default snapshot', () => {
        const { asFragment } = render(
            <PillSkeleton />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('Pill indicator snapshot', () => {
        const { asFragment } = render(
            <PillSkeleton variant="indicator" />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('Pill animate snapshot', () => {
        const { asFragment } = render(
            <PillSkeleton animate={true} />
        )

        expect(asFragment()).toMatchSnapshot()
    })
})