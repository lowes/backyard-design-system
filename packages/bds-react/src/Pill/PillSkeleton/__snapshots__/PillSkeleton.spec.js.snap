// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill Snapshots Pill animate snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  width: calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2));
  height: calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2));
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect {
  width: calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2));
  height: calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2));
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 variant--circle shape--rounded pill--skeleton variant--pill style--animated"
    height="calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2))"
    width="calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2))"
  />
</DocumentFragment>
`;

exports[`Pill Snapshots Pill default snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  width: calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2));
  height: calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2));
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect {
  width: calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2));
  height: calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2));
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 variant--circle shape--rounded pill--skeleton variant--pill style--default"
    height="calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2))"
    width="calc(var(--bds-sizes-size-24) + var(--bds-sizes-size-2))"
  />
</DocumentFragment>
`;

exports[`Pill Snapshots Pill indicator snapshot 1`] = `
<DocumentFragment>
  .c0 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c0.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c0.variant--circle {
  width: var(--bds-sizes-size-18);
  height: var(--bds-sizes-size-18);
  border-radius: var(--bds-border-radius-circle);
}

.c0.variant--rect {
  width: var(--bds-sizes-size-18);
  height: var(--bds-sizes-size-18);
}

.c0.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--rect.shape--squared {
  border-radius: 0;
}

.c0.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c0.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c0.variant--text.shape--squared {
  border-radius: 0;
}

<div
    class="c0 variant--circle shape--rounded pill--skeleton variant--indicator style--default"
    height="var(--bds-sizes-size-18)"
    width="var(--bds-sizes-size-18)"
  />
</DocumentFragment>
`;
